### PR TITLE
Leaderboard

### PR DESCRIPTION
**Pull request by AACHOURMOHAMED**
### Leaderboard

> Add get API function
> Add refereshe function
> Add post function

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
-  item1.className = 'itemScore';
-  item1.textContent = index.score;
-  List.appendChild(item1);
-});
\ No newline at end of file
+add.addEventListener('click', () => {
+  const newItemIndex = listItem.length;
+  if (username.value && score.value && score.value <= 9999999) {
+    addScoreToApi(username.value, score.value);
+    List(username.value, score.value, newItemIndex);
```
Comment (FrederickMih) [inline src/index.js]: - [ ] Based on the requirements of this project, the score list should only change when the refresh button is clicked and not immediately when the submit button is clicked. You may raise this line of code when you called the click event listener function on the refresh button.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @AACHOURMOHAMED   , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




# To Highlights

- GitHub workflow has been followed :heavy_check_mark: 
- PR has a title and summary :heavy_check_mark: 
- Linter checks are Ok :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi  @AACHOURMOHAMED , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights

✔️ No linter errors
✔️ Github flow was correctly used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (AACHOURMOHAMED) [reply src/index.js]: > Thank you for the review 
> best

**Comment by AACHOURMOHAMED** [comment]
Thank you for approving the project @danmainah 
best

**Pull request merged**